docs/source/setup.md

```diff
@@ -75,6 +75,8 @@ app.use(
 );
 ```
 
+**NOTE:** The value passed to `context` can also be a function. However, [no arguments are passed when it’s called](https://github.com/apollographql/apollo-server/blob/df51fd90dad90c9468a74e2c4ec2a6af69f4188d/packages/apollo-server-core/src/runHttpQuery.ts#L138-L139), so there’s not really any reason to do so.
```

Comment:
There is actually a reason for this: Constructing a new context based on the user request.

Reply:
Exactly like in this scenario https://github.com/apollographql/apollo-server/issues/420#issuecomment-307495466